src/lib.rs

```diff
@@ -142,6 +142,24 @@ impl<T> Vec<T> {
         self.raw.push(value)
     }
 
+    /// Appends the element returned from the closure `f` to the back of the vector
+    /// at the index supplied to the closure.
+    ///
+    /// Returns the index that the element was inserted into.
+    /// # Examples
+    ///
+    /// ```
+    /// let vec = boxcar::vec![0, 1];
+    /// vec.push_with(|index| index + 1);
+    /// assert_eq!(vec, [0, 1, 2]);
```

Comment:
I think this is incorrect? It was inserted into index `2`, so the value of the element should be `3`.

Reply:
Oh boy, i goofed that entirely. I can't even begin to explain what happened there cause I have no idea.

I *think* I was trying to figure out what the debug output of `(_, _)` was, added `+ 1` there and then promptly forgot about it (the `+ 1`) and `- 1` to get it to fire.

ugh, that's quiet embarrassing. Sorry about that. It should be fixed now.